src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,397 @@
+
+                    $fileList[$file['name']] = $file['id'];
+                } else {
+                    if (!$fileId) {
+                        continue;
+                    }
+
+                    $responses[] = $this->uploadTranslations($fileId, $domain, $content, $catalogue->getLocale());
+                }
+            }
+        }
+
+        foreach ($responses as $response) {
+            if (200 !== $response->getStatusCode()) {
+                $this->logger->error(sprintf('Unable to upload translations to Crowdin. Message: "%s".', $response->getContent(false)));
+            }
+        }
+    }
+
+    public function read(array $domains, array $locales): TranslatorBag
+    {
+        $fileList = $this->getFileList();
+
+        $translatorBag = new TranslatorBag();
+        $responses = [];
+
+        foreach ($domains as $domain) {
+            $fileId = $this->getFileIdByDomain($fileList, $domain);
+
+            if (!$fileId) {
+                continue;
+            }
+
+            foreach ($locales as $locale) {
+                if ($locale !== $this->defaultLocale) {
+                    $response = $this->exportProjectTranslations($locale, $fileId);
+                } else {
+                    $response = $this->downloadSourceFile($fileId);
+                }
+
+                $responses[] = [
+                    'response' => $response,
+                    'locale' => $locale,
+                    'domain' => $domain,
+                ];
+            }
+        }
+
+        foreach ($responses as $responseData) {
+            /** @var ResponseInterface $response */
+            $response = $responseData['response'];
+
+            if (204 === $response->getStatusCode()) {
+                $this->logger->error(sprintf('No content in exported file. Message: "%s".', $response->getContent(false)));
+
+                continue;
+            }
+
+            if (200 !== $response->getStatusCode()) {
+                $this->logger->error(sprintf('Unable to export file. Message: "%s".', $response->getContent(false)));
+
+                continue;
+            }
+
+            $exportResponse = $this->filesDownloader->request('GET', $response->toArray()['data']['url']);
+
+            if (200 !== $exportResponse->getStatusCode()) {
+                $this->logger->error(sprintf('Unable to download file content. Message: "%s".', $response->getContent(false)));
+
+                continue;
+            }
+
+            $translatorBag->addCatalogue($this->loader->load($exportResponse->getContent(), $responseData['locale'], $responseData['domain']));
+        }
+
+        return $translatorBag;
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        $fileList = $this->getFileList();
+        $responses = [];
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            if ($catalogue->getLocale() !== $this->defaultLocale) {
+                continue;
+            }
```

Comment:
Instead of loop over the catalogues, you can pick the one matching the `defaultLocale` of the first one, like I did here : https://github.com/symfony/symfony/pull/40926/files#diff-5b357f0cf7c782197270b286f525746d9e9848ffaff3e24890c4747d1bbdcd8cR62-R66

Reply:
Good idea and I will do it